pkg/client/apiutil/dynamicrestmapper.go

```diff
@@ -31,6 +31,10 @@ import (
 
 // ErrRateLimited is returned by a RESTMapper method if the number of API
 // calls has exceeded a limit within a certain time period.
+//
+// Deprecated: as the dynamic RESTMapper now returns NoMatch errors directly
+// instead of RateLimited errors, it is meaningless and only kept for the sake
+// of backwards-compatibility.
 type ErrRateLimited struct {
```

Comment:
Does it really make sense to keep this and the `DelayIfRateLimited` if we don't use it anymore? The only use case I can see for those is to handle ratelimit responses from the mapper but that won't work anymore, so removing them and make ppls code fail to compile if they use it seems the better option to me here. WDYT?

Reply:
See https://github.com/kubernetes-sigs/controller-runtime/pull/1151#discussion_r495996649 :)